source/matrix_free/task_info.cc

```diff
@@ -1014,10 +1014,62 @@ namespace internal
           blocks                      = {0, comm_begin, comm_end, end};
         }
 
-      // Step 7: Fill in the data by batches for the locally owned cells.
+      // Step 7: sort ghost cells according to the category
+      std::vector<std::array<unsigned int, 2>> tight_category_map_ghost;
+
+      if (cell_vectorization_categories.empty() == false)
+        {
+          tight_category_map_ghost.reserve(n_ghost_cells);
+
+          std::set<unsigned int> used_categories;
+          for (unsigned int i = 0; i < n_ghost_cells; ++i)
+            used_categories.insert(
+              cell_vectorization_categories[i + n_active_cells]);
+
+          std::vector<unsigned int> used_categories_vector(
+            used_categories.size());
+          n_categories = 0;
+          for (const auto &it : used_categories)
+            used_categories_vector[n_categories++] = it;
+
+          std::vector<unsigned int> counters(n_categories, 0);
+
+          for (unsigned int i = 0; i < n_ghost_cells; ++i)
+            {
+              const unsigned int index =
+                std::lower_bound(
+                  used_categories_vector.begin(),
+                  used_categories_vector.end(),
+                  cell_vectorization_categories[i + n_active_cells]) -
+                used_categories_vector.begin();
+              AssertIndexRange(index, used_categories_vector.size());
+              tight_category_map_ghost.emplace_back(
+                std::array<unsigned int, 2>{{index, i}});
+
+              // account for padding in the hp and strict case
+              if (categories_are_hp || cell_vectorization_categories_strict)
+                counters[index]++;
+            }
+
+          // insert padding
+          for (unsigned int i = 0; i < counters.size(); ++i)
+            if (counters[i] % n_lanes != 0)
+              for (unsigned int j = counters[i] % n_lanes; j < n_lanes; ++j)
+                tight_category_map_ghost.emplace_back(
+                  std::array<unsigned int, 2>{
+                    {i, numbers::invalid_unsigned_int}});
```

Comment:
Shouldn't it suffice to write
```suggestion
                    i, numbers::invalid_unsigned_int);
```

Reply:
Sorry, I was wrong, this won't work. We should keep your code.

Reply:
No. This is not working.